lib/Graph/Graph.cpp

```diff
@@ -682,6 +682,38 @@ Node *Function::createChannelShuffle(llvm::StringRef name, NodeValue input,
   return createReshape(name.str() + ".reshape2", T, inDims);
 }
 
+Node *Function::createSqueeze(llvm::StringRef name, NodeValue input,
+                              llvm::ArrayRef<size_t> axes) {
+  auto originalSize = axes.size();
+  assert(originalSize > 0 && "Parameter `axes` must be provided.");
+
+  ShapeVector shapeAxes(axes.begin(), axes.end());
+
+  // Sort and unique the values in axes to 
+  // 1. check if each value is larger than 0;
+  // 2. make sure each dim is only removed once.
+  std::sort(shapeAxes.begin(), shapeAxes.end());
+  shapeAxes.erase(std::unique(shapeAxes.begin(), shapeAxes.end()),
+                  shapeAxes.end());
+  GLOW_ASSERT(shapeAxes.front() >= 0 && "Dimension ids must be non-negative.");
+
+  auto inDims = input.dims();
+  assert(
+      shapeAxes.back() < inDims.size() &&
+      "The size of dimensions to squeeze must be less than the input size");
```

Comment:
this is not "the size of dimensions", right?

Reply:
You are right. Not only the size, but also the value of each dimension to be removed should not greater than the input size.